Sage-Bionetworks/synapse-web-monorepo
# GenUI-ADO-2627: Data access request updates for Challenge Participants

## ADO Ticket:
[FEATURE 2627](https://dev.azure.com/GenUIADO/GenUI%20Sage%20Bionetworks%20-%20Challenge%20Portal/_workitems/edit/2627)

## Description:
Updates the `AccessRequirementsList` and related code to accept a `teamId` which in turn, gets used to fetch team requirements.

## Testing:
- Spin up component in Storybook and confirm that regular `AccessRequirementList` stories behave as accepted (not signed into a Synapse account, etc.) then;
- In `AccessRequirementList.tsx`:
  - Set prop `teamId` to default value of `'3445884'`
  - Set prop `requestObjectName` to default value of `'Awesome challenge'`
- Spin up component in Storybook and confirm screenshots below for not accepted yet, accepted terms, etc. 

## Screenshots:
Figma design:
<br>
![Screenshot 2023-05-12 at 11 24 27 AM](https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/8021136/4728a997-08c4-4552-b512-99e0b1f7784e)


<br><br><br>

Prior to accepting terms of the "Awesome Challenge" that are fetched from [team access requirements endpoint](https://rest-docs.synapse.org/rest/GET/team/id/accessRequirement.html):
<br>
![Screenshot 2023-05-12 at 10 14 13 AM](https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/8021136/c7528e8b-82b0-4eb1-a5a2-f664b84bb3ee)
<br><br>

After acceptance of terms with "View Terms" visible:
<br>
![Screenshot 2023-05-12 at 10 34 21 AM](https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/8021136/e35f643a-1d02-4a5d-b22c-b485b01394d3)
<br><br>

After acceptance of terms with "Hide Terms" visible:
![Screenshot 2023-05-12 at 10 34 26 AM](https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/8021136/f3ae1668-eb3d-4cba-85ea-b7ca352283a2)
